primitives/src/secondary_key.rs

```diff
@@ -142,6 +154,24 @@ impl Permissions {
             iter::once(pallet_permissions).collect(),
         ));
     }
+
+    /// Returns the complexity of the permissions.
+    pub fn complexity(&self) -> usize {
+        // Calculate the pallet/extrinsic permissions complexity cost.
+        let cost = self.extrinsic.inner().map_or(0usize, |set| {
+            set.iter()
+                .fold(0, |cost, pallet| cost.saturating_add(pallet.complexity()))
+        });
```

Comment:
Can you add `fn fold<B>(&self, A, impl FnMut(B, &A) -> B) -> B, ` on `SubsetRestriction` and refactor the two cases here?